services/webhook-handler-active-object.ts

```diff
@@ -0,0 +1,134 @@
+import { queryToDify } from "./dify";
+import { queryToRasa } from "./rasa";
+import { markChatAsRead, sendChatbotReply } from "./whatsapp";
+import amqp from "amqplib";
+
+const { CONNECTION_PLATFORM, RABBIT_MQ_URL } = process.env;
+const DIFY = "dify";
+const RASA = "rasa";
+
+
+export interface WebhookHandlerActiveObject {
+  handleWebhookCall(body: unknown):Promise<void>;
+  init():Promise<void>;
+}
+
+const QUEUE_NAME = "WEBHOOK_QUEUE";
+export class WebhookHandlerUsingRabbitMQ implements WebhookHandlerActiveObject {
+
+
+  constructor(){
+    
+  }
+  init(): Promise<void> {
+    return this.consumeQueue();
+  }
+
+  async consumeQueue(): Promise<void> {
+    try {
+      console.info(`Connecting to ${RABBIT_MQ_URL}...`)
+      const connection = await amqp.connect(RABBIT_MQ_URL || "");
+      console.info(`Queue connected to ${RABBIT_MQ_URL}`);
+
+      const channel = await connection.createChannel();
+
+      process.once("SIGINT", async () => {
+        await channel.close();
+        await connection.close();
+      });
+
+      await channel.assertQueue(QUEUE_NAME, { durable: false });
+      await channel.consume(
+        QUEUE_NAME,
+        (message) => {
+          if (message) {
+            this.queryChatbot(JSON.parse(message.content.toString()));
+          }
+        },
+        { noAck: true }
+      );
+
+      console.log(" [*] Waiting for messages. To exit press CTRL+C");
+    } catch (err) {
+      console.warn(err);
+    }
+  }
+  async handleWebhookCall(req: Record<string, unknown>): Promise<void> {
```

Comment:
to make it modular, it should be coded as a callable function and just be called like this:

```javascript
// services/queue.ts

function sendToQueue(queue, message) {
  return new Promise((resolve, reject) => {
    amqp.connect('amqp://localhost', (error0, connection) => {
      if (error0) {
        return reject(error0);
      }
      connection.createChannel((error1, channel) => {
        if (error1) {
          return reject(error1);
        }

        channel.assertQueue(queue, {
          durable: false
        });

        channel.sendToQueue(queue, Buffer.from(message));
        console.log(" [x] Sent %s", message);

        setTimeout(() => {
          connection.close();
          resolve();
        }, 500);
      });
    });
  });
}
```

```javascript
// api/webhook.ts

const { sendToQueue } = require('./send');

const queue = 'hello';
const message = 'Hello World from!'; // json stringify if data is JSON

await sendToQueue(queue, message);
```